webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
+    } else if ("renamePage".equals(action)) {
+      // RENAME PAGE -------------------------------------------------------------
+      // = null la premiere fois, puis = nom du repertoire courant
+      String name = request.getParameter("name");
+      // = changement de noms des fichiers et repertoires
+      String newName = request.getParameter("newName");
+
+          /* Modifier le nom du fichier */
+      scc.renameFile(currentPath, name, newName);
+
+      request.setAttribute("Site", scc.getSessionSite());
+    } else if ("deletePage".equals(action)) {
+      // DELETE PAGE -------------------------------------------------------------
+      // = null la premiere fois, puis = nom du repertoire courant
+      String name = request.getParameter("name");
+
+          /* Supprimer la page */
+      scc.deleteFile(currentPath + "/" + name);
+
+      request.setAttribute("Site", scc.getSessionSite());
+    } else if ("classifySite".equals(action)) { // cas de l'upload et du design
+      // CLASSIFY SITE -------------------------------------------------------------
+      // = en cas de new Site ou de classifySite
+      String listeTopics = request.getParameter("ListeTopics");
+
+      request.setAttribute("Site", scc.getSessionSite());
+
+          /* publications : classer le site dans les themes cochés */
+      ArrayList<String> arrayToClassify = new ArrayList<>();
+      boolean publish = false;
+
+      ArrayList<String> arrayTopic = new ArrayList<>();
+      int begin = 0;
+      int end = 0;
+      end = listeTopics.indexOf(',', begin);
+      String idTopic = null;
+      while (end != -1) {
+        idTopic = listeTopics.substring(begin, end);
+
+        begin = end + 1;
+        end = listeTopics.indexOf(',', begin);
+
+        arrayTopic.add(idTopic);
+        publish = true;
+      }
+      scc.updateClassification(id, arrayTopic);
+
+      arrayToClassify.add(id);
+      if (publish) {
+        scc.publish(arrayToClassify); // set etat du site a 1
+      } else {
+        scc.dePublish(arrayToClassify); // set etat du site a 0
+      }
+    } else if ("design".equals(action)) {
+      // DESIGN -------------------------------------------------------------
+      SiteDetail site = scc.getWebSite(id);
+      scc.setSessionSite(site);
+      request.setAttribute("Site", site);
+    } else {
+      // AUTRE -------------------------------------------------------------
+      // view en cas de rechargement de la page pour naviguer dans le chemin
+      // ou createSite annule
+      // ou upload d'image
+
+      SiteDetail site = scc.getSessionSite();
+      id = site.getPK().getId();
+      request.setAttribute("Site", site);
+
+    }
+
+    return "/webSites/jsp/design.jsp?Action=design&path=" + currentPath + "&Id=" + id;
+  }
+
+  private String manageSite(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "userRole". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)